### PR TITLE
chore: upgrade ajv to the latest version 7 release

### PR DESCRIPTION
**Pull request by nigelhanlon**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
please remove package-lock.

File: test/make-test.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ function makeTest (t, options, isOk, confExpected, errorMessage) {
     t.strictSame(conf, confExpected)
   } catch (err) {
     if (isOk) {
+      console.log(err)
```
Comment (jsumners) [inline test/make-test.js:14]: Why this `console.log`?

Comment (nigelhanlon) [reply test/make-test.js:14]: That was me debugging the errors raised by the test. There are still some failures that I am trying to resolve.

Comment (nigelhanlon) [reply test/make-test.js]: Removed now.

File: .gitignore [hunk 2]
```diff
@@ -59,3 +59,6 @@ typings/
 
 # next.js build output
 .next
+
+# ignore package lock
+package-lock.json
```
Comment (simoneb) [inline .gitignore:64]: nit: add blank line

File: index.js [hunk 3]
```diff
@@ -13,9 +10,17 @@ ajv.addKeyword('separator', {
...
+    if (parentData && parentDataProperty) {
+      parentData[parentDataProperty] = data === '' ? [] : data.split(schema)
+    } else {
+      const {
+        parentData: pData,
+        parentDataProperty: pDataProperty
+      } = dataPath
+      pData[pDataProperty] = data === '' ? [] : data.split(schema)
+    }
   }
```
Comment (simoneb) [inline index.js:22]: can you explain what this does and why it's needed?

File: index.js [hunk 4]
```diff
@@ -34,6 +39,19 @@ const optsSchema = {
     expandEnv: { type: ['boolean'], default: false }
   }
 }
+
+const Ajv = require('ajv').default
```
Comment (simoneb) [inline index.js:43]: let's stick to the convention of having requires at the top

**Review by simoneb — CHANGES_REQUESTED** [review]

File: index.js (hunk 3 above)
Comment (nigelhanlon) [reply index.js:26]: In some cases, `parentData` and `parentDataProperty` will be undefined. We need to fall back to the `dataPath` object to provide those values. I have added a comment to the above.

Comment (simoneb) [reply index.js:26]: I'm curious how you found this out. Are there tests failing if you don't have this fallback in place? And do you know exactly when it happens?

Comment (nigelhanlon) [reply index.js:26]: Without the fall back there are three failing tests. I added a console logger to inspect the properties being passed to the separator compile function and noted that the parameters used were undefined. 

I added a conditional check to make sure the parent attributes were defined but this applied the separator logic incorrectly and still left failing tests. Attached below is one such test. I then went further and used the properties from the `dataPath` object to replace the undefined values. The tests then passed as expected. 

image

**Review by simoneb — APPROVED** [review]

**Comment by climba03003** [comment]
Should this be landed until `fastify` support `ajv@7`?

**Comment by simoneb** [comment]
> Should this be landed until `fastify` support `ajv@7`?

I don't think so, this can be used independently of fastify and the upgrade in fastify may be slow to come. Unless there are other reasons to delay this upgrade we should merge.

**Comment by climba03003** [comment]
> I don't think so, this can be used independently of fastify and the upgrade in fastify may be slow to come. Unless there are other reasons to delay this upgrade we should merge.

Make sense. Then, it should be bumping a `major` release as it may break the usage when the user is still on `ajv@6`.

**Comment by simoneb** [comment]
I don't think it should be a major, ajv is internal to this library and if we've done the upgrade right there's not breaking change exposed to the user.

**Review by climba03003 — APPROVED** [review]
After the research on how `npm` and `yarn` for dependencies resolution. They are smarter than I though. I have no more question about the `version` or `delay of landing`.

**Review by simoneb — APPROVED** [review]

**Comment by simoneb** [comment]
@mcollina @jsumners this is ready for review

**Review by jsumners — APPROVED** [review]
LGTM

**Pull request merged**